### ISSUE TITLE
When creating a ticket contracts shown are not all available

### ISSUE THREAD

**Issue by OscarBeiro** [glpi-project/glpi#17208]
### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Version

10.0.15

### Bug description

When you create a new ticket, the maximum number of items in the contract limits the number of contracts linked to it.
Once you create the ticket, you can use the contract tab to add it, but it is still a bug.

We are providing a PR.

### Relevant log output

```shell
Not relevant
```

### Page URL

ticket.form.php

### Steps To reproduce

1. Create two contracts in an subentity
2. both of them will have Max number of items = 1
3. Create a new ticket
4. Try to add the second contract. It won't be shown.

### Your GLPI setup information

Tested on a clean GLPI 10.0.15

### Anything else?

I am creating it here because the portal is down for maintenance

**Comment by FrancisTicgal** [comment]
Tested on newly installed GLPI 10.0.15.
![image](https://github.com/glpi-project/glpi/assets/114425698/e90ee83d-168d-4c53-b644-9d7d356f364c)

The contract dropdown in ticket creation is being limited by the number of assets associated with the contracts, causing those with no more relationships available not to be displayed.

Once the ticket is created, within the contracts tab they all appear, which we understand that during ticket creation should have the same behavior.
![image](https://github.com/glpi-project/glpi/assets/114425698/78209420-9d43-40c5-b9ab-9cecd606ffb2)

The solution is to correct the twig layout template.
https://github.com/glpi-project/glpi/blob/7a20a9ae91fc985337d4582d64aec70a1d111cb6/templates/components/itilobject/fields_panel.html.twig#L230

```twig
            {% if item.isNewItem() and item.getType() == 'Ticket' %}
               {{ fields.dropdownField(
                  'Contract',
                  '_contracts_id',
                  params['_contracts_id']|default(0),
                  'Contract'|itemtype_name,
                  field_options|merge({
                     'entity': item.fields['entities_id'],
                     'width': '100%',
                     'hide_if_no_elements': true,
                     'nochecklimit': true
                  })
               ) }}
            {% endif %}
```
Adding the nochecklimit option so that it does not check the number of relationships with items, since relationships with tickets are stored in glpi_tickets_contracts and not in glpi_contracts_items.

**Comment by cconard96** [comment]
Fixed by #17210
